### PR TITLE
#563 Make ReaderAsBytes close the Reader on calling Bytes#asBytes()

### LINKED ISSUE #563: ReaderAsBytes not closing the enclosed Reader
`ReaderAsBytes.asBytes()` consumes the enclosed reader but does not close it. This does not make any sense because there is nothing more to read from the Reader, and also will result in some unexpected behavior in some environments.

For example, in Windows, attempt to read the bytes of a text file and then delete:
```java
final Path file = ...;
final byte[] contents = new ReaderAsBytes(new ReaderOf(file)).asBytes();
Files.delete(file);    //IOException "file is being used by another process"
```

### PR DESCRIPTION
**Pull request by rexim**
Close #563

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#598](https://github.com/yegor256/cactoos/issues/598) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#598](https://github.com/yegor256/cactoos/issues/598) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/java/org/cactoos/io/EmptyClosableReader.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * <p>This class is for internal use only. Use {@link ReaderOf} instead</p>
+ *
+ * <p>There is no thread-safety guarantee.</p>
+ *
+ * @author Alexey Kutepov (reximkut@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ */
+public final class EmptyClosableReader extends Reader {
```
Comment (neonailol) [inline src/main/java/org/cactoos/io/EmptyClosableReader.java]: please remove `public` from this class, as it is internal

**Comment by rexim** [comment]
@neonailol done. Please take a look again when you get a chance.

File: src/main/java/org/cactoos/io/EmptyClosableReader.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (yegor256) [inline src/main/java/org/cactoos/io/EmptyClosableReader.java]: @rexim I think this class should be either inner class of `ReaderAsBytesTest` or at least stay in `src/test`

**Comment by yegor256** [comment]
@rexim see above

File: src/main/java/org/cactoos/io/EmptyClosableReader.java (hunk 2 above)
Comment (rexim) [reply src/main/java/org/cactoos/io/EmptyClosableReader.java]: @yegor256 I agree. `org.cactoos.io` in `src/test` already contains a couple of "Testing Helper Classes" like `SlowInput` or  `SlowInputStream`. I'd go for moving `EmptyClosableReader` there. I have a feeling that this class might be useful for testing similar `Reader` consumers in the future.

**Comment by rexim** [comment]
@yegor256 addressed the remark. Please take a look again when you get a chance.

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol), total is [+330](https://www.0crat.com/u/neonailol)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9295](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#598](https://github.com/yegor256/cactoos/issues/598) is now out of scope

**Pull request merged**